runtime/type_test.go

```diff
@@ -90,3 +90,48 @@ func TestRuntimeTypeStorage(t *testing.T) {
 
 	assert.Equal(t, `"Int"`, loggedMessage)
 }
+
+
+func TestBlockTimestamp(t *testing.T) {
+	t.Parallel()
+	runtime := NewInterpreterRuntime()
+	script := []byte(`
+		transaction {
+			prepare() {
+				let block = getCurrentBlock()
+				var ts: UFix64 = block.timestamp
+				log(ts.isInstance(Type<UFix64>()))
+
+				var div: UFix64 = 4.0
+				var x = ts as UFix64
+
+				// Shouldn't panic
+				var result = ts/div
+			}
+		}
```

Comment:
Maybe make this a script, i.e. replace this with a `main` function and call `ExecuteScript` below, so it can return a value, and then return an array which contains both `ts` and `result`.

At the bottom of the test you can then assert that all elements are actual `interpreter.UFix64Value` Go values (you can use `assert.IsType`)

Reply:
When running as a script `getCurrentBlock()` seems to return a nil reference resulting in the following error: 
```
panic: runtime error: invalid memory address or nil pointer dereference
```
Not sure whether that is expected though.